dotnet/src/SemanticKernel.Test/Memory/VolatileMemoryStoreTests.cs

```diff
@@ -248,4 +248,35 @@ public async Task GetNearestAsyncReturnsExpectedWithMinScoreAsync()
             Assert.True(compare >= 0);
         }
     }
+
+    [Fact]
+    public async Task GetNearestAsyncDifferentiatesIdenticalVectorsByKeyAsync()
+    {
+        // Arrange
+        var compareEmbedding = new Embedding<double>(new double[] { 1, 1, 1 });
+        int rand = Random.Shared.Next();
+        string collection = "collection" + rand;
+        int topN = 4;
+
+        string key = "key" + Random.Shared.Next();
+        var memory = new DoubleEmbeddingWithBasicMetadata(new Embedding<double>(new double[] { 1, 1, 1 }), "1 ,2 ,3");
+        await this._db.PutValueAsync(collection, key, memory);
+
+        for (int i = 0; i < 10; i++)
+        {
+            key = "key" + Random.Shared.Next();
+            memory = new DoubleEmbeddingWithBasicMetadata(compareEmbedding, "1 ,1 ,1");
+            await this._db.PutValueAsync(collection, key, memory);
+        }
+
+        // Act
+        var topNResults = this._db.GetNearestMatchesAsync(collection, compareEmbedding, limit: topN, minRelevanceScore: 0.75).ToEnumerable().ToArray();
+
+        // Assert
+        for (int i = 0; i < topNResults.Length; i++)
```

Comment:
nit: assert Length is topN

Reply:
+1

I suspect this test is going to fail with how it's currently written

Reply:
Resolved with latest commits, thanks :)